### ISSUE TITLE
1.6.0-rc1/java compile error

### ISSUE THREAD

**Issue by tonycurtis** [openucx/ucx#3651]
make failure on standalone machine with java enabled, out-of-tree-build:

```shell
Making install in bindings/java/src/main/native
make[1]: Entering directory `/usr/local/src/ucx/build/bindings/java/src/main/native'
mvn -B -f ../../../../../../ucx-1.6.0/bindings/java/pom.xml -Dmaven.repo.local=/usr/local/src/ucx/build/.deps/ compile native:javah
POM file ../../../../../../ucx-1.6.0/bindings/java/pom.xml specified with the -f/--file command line argument does not exist
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-readable POM /usr/local/src/ucx/build/bindings/java/src/main/native/../../../../../../ucx-1.6.0/bindings/java/pom.xml: /usr/local/src/ucx/build/bindings/java/src/main/native/../../../../../../ucx-1.6.0/bindings/java/pom.xml (No such file or directory) @
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project  (/usr/local/src/ucx/build/bindings/java/src/main/native/../../../../../../ucx-1.6.0/bindings/java/pom.xml) has 1 error
[ERROR]     Non-readable POM /usr/local/src/ucx/build/bindings/java/src/main/native/../../../../../../ucx-1.6.0/bindings/java/pom.xml: /usr/local/src/ucx/build/bindings/java/src/main/native/../../../../../../ucx-1.6.0/bindings/java/pom.xml (No such file or directory)
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
make[1]: *** [org_ucx_jucx_ucp_UcpContext.h] Error 1
make[1]: Leaving directory `/usr/local/src/ucx/build/bindings/java/src/main/native'
make: *** [install-recursive] Error 1
```

**Comment by petro-rudenko** [comment]
Can you please provide steps how did you run configure?

**Comment by tonycurtis** [comment]
Out of tree build, simplest configure that causes problems is:

```shell
../ucx-1.6.0/configure --with-java=/opt/oracle/jdk/jdk-9.0.4
```

**Comment by petro-rudenko** [comment]
```
 git clone https://github.com/openucx/ucx.git ucx-1.6.0
mkdir out-of-tree-build
cd ucx-1.6.0
./autogen.sh
cd ../out-of-tree-build
../ucx-1.6.0/configure --with-java
make 
```
works for me.

Does it generate `pom.xml` in bindings/java ?

**Comment by tonycurtis** [comment]
Same problem in a fresh build.

But this is the 1.6.0 rc1 release (not cloned), so there's no autogen.

**Comment by petro-rudenko** [comment]
Ok so release tarball doesn't contains java sources.

**Comment by yosefe** [comment]
@petro-rudenko which stage should generate the missing java sources?

**Comment by shamisp** [comment]
This issue was reported on v1.6.x branch and the patch was merged into the master. Re-opening the ticket. @petro-rudenko do we have PR for v1.6.0 ?

**Comment by yosefe** [comment]
Fixed for v1.6.x in #3667
